src/Coordination/Changelog.cpp

```diff
@@ -142,8 +142,12 @@ class ChangelogWriter
 
 struct ChangelogReadResult
 {
+    /// Total entries read from log including skipped
     uint64_t entries_read;
+    /// First entry actually read log (not including skipped)
     uint64_t first_read_index;
+    /// Last entry read from log
+    uint64_t last_read_index;
```

Comment:
How it differs from `entries_read`? If indexes starts with zero and incrementally increased then `last_read_index` should equals to `entries_read - 1`, it's unclear to me from comments why it's not.

Reply:
It's really confusing logic and I don't know how to avoid it. We just have a lot of possible cases of our logs state. 

We have to read from log starting from some `offset_index` and result is three numbers:
1) `entries_read` -- how many entries were in log ignoring `offset_index`. This number is useful for some checks like `log is in consistent state` or `log not finished yet` and so on. 
2) `first_read_index` -- first index we have read starting from `offset_index`. In normal case it should be equal to `offset_index`, but sometimes not.
3) `last_read_index` -- last read index from log.
So if our `offset_index` equal to zero (or to the index of the first entry in log) we will have `last_read_index = first_read_index + entries_read`. But in case of non zero `offset_index` we will have different values.